scalding-serialization/src/main/scala/com/twitter/scalding/serialization/macros/impl/ordered_serialization/providers/ProductOrderedBuf.scala

```diff
@@ -42,54 +48,182 @@ object ProductOrderedBuf {
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any]],
+      typeOf[
+        Product21[Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any]],
+      typeOf[
+        Product22[Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any,
+                  Any]]
+    )
 
     def validType(curType: Type): Boolean =
-      validTypes.exists { t => curType <:< t }
+      validTypes.exists { t =>
+        curType <:< t
+      }
 
     // The `_.get` is safe since it's always preceded by a matching
     // `_.isDefined` check in `validType`
     @SuppressWarnings(Array("org.wartremover.warts.OptionPartial"))
     def symbolFor(subType: Type): Type = {
-      val superType = validTypes.find{ t => subType.erasure <:< t }.get
+      val superType = validTypes.find { t =>
+        subType.erasure <:< t
+      }.get
       subType.baseType(superType.typeSymbol)
     }
 
     val pf: PartialFunction[c.Type, TreeOrderedBuf[c.type]] = {
-      case tpe if validType(tpe.erasure) => ProductOrderedBuf(c)(buildDispatcher, tpe, symbolFor(tpe))
+      case tpe if validType(tpe.erasure) =>
+        ProductOrderedBuf(c)(buildDispatcher, tpe, symbolFor(tpe))
     }
     pf
   }
 
-  def apply(c: Context)(buildDispatcher: => PartialFunction[c.Type, TreeOrderedBuf[c.type]], originalType: c.Type, outerType: c.Type): TreeOrderedBuf[c.type] = {
+  def apply(c: Context)(buildDispatcher: => PartialFunction[c.Type, TreeOrderedBuf[c.type]],
+                        originalType: c.Type,
+                        outerType: c.Type): TreeOrderedBuf[c.type] = {
     import c.universe._
-    def freshT(id: String) = newTermName(c.fresh(id))
+    def freshT(id: String) = TermName(c.freshName(id))
 
     val dispatcher = buildDispatcher
     val elementData: List[(c.universe.Type, TermName, TreeOrderedBuf[c.type])] =
-      outerType
-        .declarations
+      outerType.declarations
         .collect { case m: MethodSymbol => m }
-        .filter(m => m.name.toString.startsWith("_"))
+        .filter(m => m.name.toTermName.toString.startsWith("_"))
```

Comment:
I had to remove a bunch of these since they were calling `toTermName` on things that were already `TermName`, but I guess with `blackbox` this changed?

Reply:
I suspect i've glossed over some of those warnings migrating in bulk the code with the fixes in it. I'll do a clean compile or two to make sure i've not added new warnings.